csm/core/services/sessions.py

```diff
@@ -410,3 +414,30 @@ async def delete_all_sessions_for_user(self, user_id: str) -> None:
         for each_session in session_data:
             if each_session.credentials.user_id.lower() == user_id.lower():
                 await self._session_manager.delete(each_session.session_id)
+
+    async def update_session_expiry_time(self, session: Session) -> None:
+        session.expiry_time = self._session_manager.calc_expiry_time()
+        await self._session_manager.update(session)
+
+    async def _is_valid_session_exists(self, user_id: str) -> None:
+        """
+        This Function will get the current user's active session.
+        :param user_id: user_id for user. :type:Str
+        :return: None
+        """
+        Log.debug(f"get active sessions for Userid: {user_id}")
+        session_data = await self._session_manager.get_all()
```

Comment:
One user will have only one session. Why are you getting all?

Reply:
In consul, sessions are stored in mapping of session_id to session model and we want to get sessions with user id
Algo: 
1. get all sessions() -> session list
2. traverse over all session list and Check user id in that each session, if user id matches then check if it is expired, if yes delete it else return the same session